sdk/eventhub/Azure.Messaging.EventHubs/samples/Program.cs

```diff
@@ -74,30 +229,119 @@ public static async Task Main(string[] args)
+        /// </summary>
+        ///
+        /// <param name="parsedArgs">The arguments passed from console.</param>
+        ///
+        private static void PromptConnectionString(CommandLineArguments parsedArgs)
+        {
+            // Prompt for the connection string, if it wasn't passed.
 
-            if (choice == null)
+            while (String.IsNullOrEmpty(parsedArgs.ConnectionString))
             {
+                Console.Write("Please provide the connection string for the Event Hubs namespace that you'd like to use and then press Enter: ");
+                parsedArgs.ConnectionString = Console.ReadLine().Trim();
                 Console.WriteLine();
-                Console.WriteLine();
-                Console.WriteLine("Quitting...");
-                Console.WriteLine();
-                return;
             }
-            else
+        }
+
+        /// <summary>
+        ///   Prompt the user to insert the EventHubs fully qualified name, if not
+        ///   already passed it from command line.
+        /// </summary>
+        ///
+        /// <param name="parsedArgs">The arguments passed from console.</param>
+        ///
+        private static void PromptEventHubName(CommandLineArguments parsedArgs)
+        {
+            // Prompt for the Event Hub name, if it wasn't passed.
+
+            while (String.IsNullOrEmpty(parsedArgs.EventHub))
             {
+                Console.Write("Please provide the name of the Event Hub that you'd like to use and then press Enter: ");
+                parsedArgs.EventHub = Console.ReadLine().Trim();
                 Console.WriteLine();
+            }
+        }
+
+        /// <summary>
+        ///   Prompt the user to insert the AAD service client id, if not
+        ///   already passed it from command line.
+        /// </summary>
+        ///
+        /// <param name="parsedArgs">The arguments passed from console.</param>
+        ///
+        private static void PromptClientId(CommandLineArguments parsedArgs)
+        {
+            // Prompt for the Event Hub name, if it wasn't passed.
+
+            while (String.IsNullOrEmpty(parsedArgs.ClientId))
+            {
+                Console.Write("Please provide the Azure Active Directory client identifier of the service principal: ");
+                parsedArgs.ClientId = Console.ReadLine().Trim();
                 Console.WriteLine();
+            }
+        }
+
+        /// <summary>
+        ///   Prompt the user to insert the EventHubs fully qualified namespace, if not
+        ///   already passed it from command line.
+        /// </summary>
+        ///
+        /// <param name="parsedArgs">The arguments passed from console.</param>
+        ///
+        private static void PromptFullyQualifiedNamespace(CommandLineArguments parsedArgs)
+        {
+            // Prompt for the Fully Qualified Namespace, if it wasn't passed.
+
+            while (String.IsNullOrEmpty(parsedArgs.FullyQualifiedNamespace))
```

Comment:
nit:  The coding standards request that we use `string`, despite my preference for this style.    Looks like our analysis tool missed changing these during its last pass.  Do you mind doing a global replace to `string.IsNullOrEmpty` ?